### PR TITLE
update readme to announce annotate view

### PR DESCRIPTION
**Pull request by lievenhey**
fixes: #336

### CODE REVIEW

**Comment by milianw** [comment]
I don't think that we have a proper source annotation just yet, personally. it's getting there, but is still miles away from what I would call "proper" :D

**Comment by milianw** [comment]
have a look at vtune's source annotation to understand what I mean by "proper". also look at perf's annotation view capabilities wrt following jumps etc.

File: src/models/disassemblymodel.cpp [hunk 1]
```diff
@@ -47,6 +61,14 @@ QVariant DisassemblyModel::headerData(int section, Qt::Orientation orientation,
     if (section == DisassemblyColumn)
         return tr("Assembly");
 
+    if (section == LinkedFunctionName) {
+        return QStringLiteral("Linked Function Name");
```
Comment (milianw) [inline src/models/disassemblymodel.cpp:65]: tr

File: src/models/disassemblymodel.cpp [hunk 2]
```diff
@@ -47,6 +61,14 @@ QVariant DisassemblyModel::headerData(int section, Qt::Orientation orientation,
...
+    }
+
+    if (section == LinkedFunctionOffset) {
+        return QStringLiteral("Linked Function Offset");
```
Comment (milianw) [inline src/models/disassemblymodel.cpp:69]: tr

File: src/models/disassemblyoutput.cpp [hunk 3]
```diff
@@ -22,9 +22,10 @@ static QVector<DisassemblyOutput::DisassemblyLine> objdumpParse(const QByteArray
     QString asmLine;
     // detect lines like:
     // 4f616: 84 c0 test %al,%al
-    static const QRegularExpression disassemblyRegex(QStringLiteral("^    ([0-9a-f]{4,}):\t[0-9a-f ]{2,}"));
+    static const QRegularExpression disassemblyRegex(QStringLiteral("^[ ]+([0-9a-f]{4,}):\t[0-9a-f ]{2,}"));
     while (stream.readLineInto(&asmLine))
     {
+        qDebug() << asmLine;
```
Comment (milianw) [inline src/models/disassemblyoutput.cpp:28]: remove

File: src/models/disassemblyoutput.h [hunk 4]
```diff
@@ -17,6 +17,8 @@ struct DisassemblyOutput
     {
         quint64 addr = 0;
         QString disassembly;
+        QString linkedFunctionName;
+        int linkedFunctionOffset;
```
Comment (milianw) [inline src/models/disassemblyoutput.h:21]: = 0;

also document what this offset means, it's probably the offset into the `disassembly` string, or?

File: src/models/disassemblyoutput.cpp [hunk 5]
```diff
@@ -49,9 +53,31 @@ static QVector<DisassemblyOutput::DisassemblyLine> objdumpParse(const QByteArray
                 qWarning() << "unhandled asm line format:" << asmLine;
                 continue;
             }
+
+            const auto leftBracketIndex = asmLine.indexOf(QLatin1Char('<'));
```
Comment (milianw) [inline src/models/disassemblyoutput.cpp:57]: extract this code out into a separate function then call it below, i..e something like

```
struct LinkedFunction
{
    QString name;
    int offset = 0;
};

LinkedFunction extractLinkedFunction(const QString &disassembly);
...
disassemblyLines.push_back({addr, asmLine, extractLinkedFunction(asmLine)});
```

File: src/models/disassemblyoutput.cpp [hunk 6]
```diff
@@ -40,6 +41,9 @@ static QVector<DisassemblyOutput::DisassemblyLine> objdumpParse(const QByteArray
             continue;
         }
 
+        QString linkedFunctionName;
```
Comment (milianw) [inline src/models/disassemblyoutput.cpp:44]: remove (see below)

File: src/resultspage.cpp [hunk 7]
```diff
@@ -104,10 +104,10 @@ ResultsPage::ResultsPage(PerfParser* parser, QWidget* parent)
     m_summaryPageDock->addDockWidgetAsTab(m_disassemblyDock, KDDockWidgets::InitialVisibilityOption::StartHidden);
     m_disassemblyDock->toggleAction()->setEnabled(false);
     m_summaryPageDock->setAsCurrentTab();
-    if (m_frequencyPage) {
-        m_frequencyDock = dockify(m_frequencyPage, QStringLiteral("frequency"), tr("Fr&equency"), tr("Ctrl+E"));
-        m_summaryPageDock->addDockWidgetAsTab(m_frequencyDock);
-    }
+#if QCustomPlot_FOUND
```
Comment (milianw) [inline src/resultspage.cpp:107]: why this change?

**Review by milianw — CHANGES_REQUESTED** [review]

File: src/models/disassemblyoutput.h (hunk 4 above)
Comment (lievenhey) [reply src/models/disassemblyoutput.h:21]: it is the offset from the start of the function

File: src/resultspage.cpp (hunk 7 above)
Comment (lievenhey) [reply src/resultspage.cpp:107]: `frequencypage.h` is only included when `QCustomPlot` is found. If it is not found, the compiler doesn't know that `m_frequencyPage` is a QWidget and fails to compile.
This Change is also in #360

Comment (milianw) [reply src/resultspage.cpp:107]: ok but that has nothing to do with the topic of "update readme to announce annotate view" and should thus not be part of this pull request
